modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/ConfigurationBasedElementModelFactory.java

```diff
@@ -267,4 +536,70 @@ private void resolveWrappedElement(DslElementModel.Builder<ParameterGroupModel>
     return empty();
   }
 
+  private void handleInfrastructure(final ParameterModel paramModel, final DslElementSyntax paramDsl,
+                                    final Map<ComponentIdentifier, ComponentConfiguration> nested,
+                                    final Map<String, String> parameters,
+                                    final DslElementModel.Builder<ParameterGroupModel> groupElementBuilder) {
+
+    switch (paramModel.getName()) {
```

Comment:
this whole class has a TODO of being deleted once we have an extension model for everything, is it worth it to add technical debt?